File: mobile/lib/shared/models/event_create_response.dart
import 'event.dart';

class EventCreateResponse {
  const EventCreateResponse({
    required this.event,
    this.newToken,
    this.roleChanged = false,
  });

  factory EventCreateResponse.fromJson(Map<String, dynamic> json) {
    return EventCreateResponse(
      event: Event.fromJson(json['event'] as Map<String, dynamic>),
      newToken: json['newToken'] as String?,
      roleChanged: json['roleChanged'] as bool? ?? false,
    );
  }

  final Event event;
  final String? newToken;
  final bool roleChanged;
}

File: .idea/backend/src/main/java/com/luma/dto/response/EventCreateResponse.java
package com.luma.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EventCreateResponse {
    private EventResponse event;
    private String newToken;
    private boolean roleChanged;
}
